tensorflow/compiler/xla/service/hlo_instruction.h

```diff
@@ -1878,6 +1886,10 @@ class HloInstruction {
   // HLO. See the documentation on backend_config().
   string backend_config_;
 
+  // Attributes passed from the frontend to give hints to the backend about
+  // how to compile this HLO.
+  FrontendAttributes frontend_attributes_;
```

Comment:
We need to state that HLO->HLO transforms are expected to preserve these attributes on a "best effort" basis only.  An example would also help.

Reply:
Done